File: programmers/python/완주하지못한선수.py
def solution(participant, completion):
    answer = ''
    dict = {}
    for com in completion:
        if com in dict:
            dict[com] += 1
        else:
            dict[com] = 1
    for part in participant:
        if not(part in dict) or dict[part] == 0:
            answer = part
            break
        else:
            dict[part] -= 1
    return answer

File: programmers/c++/완주하지못한선수.cpp
//완주하지 못한 선수
#include <string>
#include <vector>
#include <map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
	string answer = "";
	map<string, int> m;
	for (auto part : participant) {
		m[part]++;
	}
	for (auto complet : completion) {
		m[complet]--;
	}
	for (auto part : participant) {
		if (m[part] > 0)
			return part;
	}
	return answer;
}
int main() {
	vector<string> participant = { "leo", "kiki", "eden" };
	vector<string> completion={ "kiki", "eden" };
	string ans = solution(participant, completion);
	cout << ans;
	return 0;
}